lib/utils/dom.js

```diff
@@ -150,16 +150,16 @@ export function scrollToElement(element, { scrollStyle, direction }) {
 		padBottom(top, viewport.height);
 		scrollTo(0, top);
 	} else if (scrollStyle === 'middle') {
-		// Do not scroll if new element position does not exceed current middle point
-		if (
-			(direction === 'up' && target.top > viewport.height / 2) ||
-			(direction === 'down' && target.bottom < viewport.height / 2)
-		) {
-			return;
+		const buffer = (viewport.height - target.height) / 2;
+		top -= buffer;
```

Comment:
Nit: use a different var for this, since it's not really the top of the target anymore